src/app/components/Common/Tracks/TracksList/TracksList.jsx

```diff
@@ -1,30 +1,22 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 
 import classes from './TracksList.scss';
 import InfiniteScroll from '../../InfiniteLoader/InfiniteScroll';
 import TrackListItem from './TracksListItem';
 
-class TracksList extends React.Component {
-  constructor(props) {
-    super(props);
+function TracksList(props) {
+  const [tracks, setTracks] = useState(null);
 
-    this.state = {
-      tracks: null,
-    };
+  function onSetItems(state) {
+    setTracks(state.items);
+    props.onSetItems(state);
   }
 
-  onSetItems = state => {
-    this.setState({
-      tracks: state.items,
-    });
-    this.props.onSetItems(state);
-  };
-
-  rowRenderer = ({ item: track, index, isScrolling, isVisible, key, style }) => {
-    const { tracks } = this.state;
-    const { showArtist, showAlbum, playTrack } = this.props;
+  const { showArtist, showAlbum, scrollElement, load, scrollElementModifier } = props;
 
+  // TODO: Fix below ESLint warnings...
+  function rowRenderer({ item: track, index, isScrolling, isVisible, key, style }) {
```

Comment:
@raphaelvigee, any clue how to fix this nicely other than removing the deconstruction from the parameter?

Reply:
Eh, moved to use deconstruction outside of the function header as a resolution. No clue why, but prop-types doesn't seem to be a fan of functional components. That's an issue for another day.